node/src/components/contract_runtime.rs

```diff
@@ -386,16 +387,37 @@ impl ContractRuntime {
             }
             ContractRuntimeRequest::GetAddressableEntity {
                 state_root_hash,
-                key,
+                entity_addr,
                 responder,
             } => {
                 trace!(?state_root_hash, "get addressable entity");
                 let metrics = Arc::clone(&self.metrics);
                 let data_access_layer = Arc::clone(&self.data_access_layer);
                 async move {
                     let start = Instant::now();
-                    let request = AddressableEntityRequest::new(state_root_hash, key);
+                    let entity_key = match entity_addr {
+                        EntityAddr::SmartContract(_)|EntityAddr::System(_) => Key::AddressableEntity(entity_addr),
```

Comment:
nitpick, formatting
```suggestion
                        EntityAddr::SmartContract(_) | EntityAddr::System(_) => Key::AddressableEntity(entity_addr),
```

Reply:
Well spotted :)

Updated in f55457f3835160b2d71e6cac255b5ba2c83764c6